src/wallet/coinselection.cpp

```diff
@@ -320,13 +323,13 @@ util::Result<SelectionResult> KnapsackSolver(std::vector<OutputGroup>& groups, c
     Shuffle(groups.begin(), groups.end(), rng);
 
     for (const OutputGroup& group : groups) {
-        if (group.GetSelectionAmount() == nTargetValue) {
```

Comment:
suggestion, add 

```cpp
if (group.m_weight <= max_input_weight) continue;
```
on top of the for loop instead of checking within each condition.

Currently, you add groups that are above weight to `applicable_groups`

Reply:
I think you mean
```cpp
if (group.m_weight > max_input_weight) continue;
```
This way we jusk skip group whose weight is larger than `max_input_weight` and avoid checking the weight limit for all the conditional branches below.